### ISSUE TITLE
Cannot add more than one listener if its expressed as a lambda

### ISSUE THREAD

**Issue by jeffdiclemente** [CppMicroServices/CppMicroServices#95]
I noticed this while implementing the framework listeners; doing the following does not add the second listener:

```
int count1(0);
int count2(0);
auto listener_callback_counter1 = [&count1](const FrameworkEvent& evt) { ++count1; std::cout << "listener_callback_counter1: call count " << count1 << std::endl; };
auto listener_callback_counter2 = [&count2](const FrameworkEvent& evt) { ++count2; std::cout << "listener_callback_counter2: call count " << count2 << std::endl; };
f->GetBundleContext()->AddFrameworkListener(listener_callback_counter1);
f->GetBundleContext()->AddFrameworkListener(listener_callback_counter2);
```

The same happens when adding bundle listeners. 
In both cases the listener comparison functions in `usServicveListeners.cpp` returns true for the second listener as compared to the first. 

If the `std::bind` flavor of `AddBundleListener` is used, there is no problem.

For users, it is normal to express std::function objects as lambdas and expect this to work.

It is difficult to compare the equality of std::function objects as evidenced by [this stackoverflow thread](https://stackoverflow.com/questions/3629835/why-is-stdfunction-not-equality-comparable). 
We'll have to define what equality really means for listeners.

**Comment by jeffdiclemente** [comment]
Some more info from further investigation...

Since ```std::function::target<void(const BundleEvent&)>()``` is used as one of the criteria to compare listeners for equality lambdas and callable objects created from ```std::bind``` will never be compared correctly because ```target<T>()``` will return a pointer only if the stored callable object is of type ```T```, otherwise it returns ```nullptr```. For lambdas and callable objects created from ```std::bind```, the type is not ```void(const BundleEvent& )```.

The only reason listeners created from ```std::bind``` can be added is because the pointer to the class instance is also used to compare the listeners. As long as two different class instance pointers are used the listeners aren't equal.

**Comment by jeffdiclemente** [comment]
fixed in PR #179
